Invert the query index matchup in CallSiamese (and update CallBokeh)

Phancisth opened the pull request:
Change log
- Modify CallSiamese.sh to invoke siamese_teddyPlus.jar (with proper parsing of input args) instead of the old siamese.jar
- Modify CallBokeh.sh to link with the new python visualization file.

Phancisth committed:
New visualization code file for teddyPlus

Completed: Extract the contents from Siamese's CSVs and rearrange them in two intermediate CSVs (idiomatic.csv, nonidiomatic.csv) for further plotting process.
To-do: Write the new plotting function

Phancisth committed:
Updating gitignore

Ignored files:
/detection-mode/idiomatic.csv
/detection-mode/nonidiomatic.csv

Phancisth committed:
Update bokeh_visualization.py

Supporting recommendation pattern, startLine and endLine information and the location of the graph located same place as the Siamese's CSVs.

Phancisth committed:
Plotting: Bug Fixes and Feature Update

Bug Fixes:
- The domain of y-axis includes lists of all files from the IP and NIP sources
- The marker circle cross, inverted triangle, and square cross changed to proper syntax

Feature Update:
- Display of recommend pattern in the hover box for NIP plots
- The output graph file is located same place as the Siamese CSVs.

Phancisth committed:
Delete temp-config.properties

Phancisth committed:
Modify the shell script for calling the new Siamese engine (for Teddy Plus)

Phancisth committed:
Update CallBokeh.sh

Modify the script to receive parse only 3 arguments (formerly 4) to the visualization script of TeddyPlus

Phancisth committed:
Update gitignore

ayanogami commented:
I checked that, I have an error
```python3: can't open file 'bokeh_visualization_teddyPlus.py': [Errno 2] No such file or directory```

I think it is only a mistake of the filename
Please chek

ayanogami committed:
show the alert if commit count is zero\n fix to get commits command

Phancisth committed:
Re-arranging the order of the log display and the method calls

Commit:
Merge pull request #46 from NAIST-SD-PBL-PAL/visualization-for-teddyplus

Visualization for Teddyplus

Commit:
Merge pull request #51 from NAIST-SD-PBL-PAL/feature/specified_interval

show the alert if commit count is zero\n fix to get commits command

ayanogami reviewed (approved):
LGTM

Phancisth committed:
New version of ES engine (patch on the output csvs naming error)

The binary of ES tends to crash easily for no reason. I fixed this problem for now by replacing the old ones with the new set of the same binaries downloaded from ES official site.

Phancisth committed:
Update CallSiamese.sh

Have the engine take in -n argument to be used for naming of the outputted CSVs (pattern is <repoName>_<trackNum>_<commitSHA>.csv

Phancisth committed:
Just adding [TeddyPlus] to the head of every terminal output log

Phancisth committed:
Merge branch 'master' into Invert-the-Query-Index-Matchup-in-CallSiamese

Phancisth commented:
So I have tested and verified that the Elastic database and the engine are connected together and CSVs can be properly generated.
